coordinatorlib/src/main/scala/com/socrata/datacoordinator/secondary/sql/PostgresSecondaryMetrics.scala

```diff
@@ -44,5 +46,55 @@ class PostgresSecondaryMetrics(conn: Connection) extends SqlSecondaryMetrics(con
           throw new Exception("Unexpected Postgres unique constraint violation!", uv)
       }
     }
+
+    val date = DateUtils.truncate(new Date(), Calendar.DATE)
+
+    val historyUpdated = using(conn.prepareStatement(
+      """UPDATE secondary_metrics_history
+            SET total_size = ?
+          WHERE store_id = ? AND dataset_system_id = ? AND date = ?
+      """)) { stmt =>
+        stmt.setLong(1, metric.totalSizeBytes)
+        stmt.setString(2, storeId)
+        stmt.setDatasetId(3, datasetId)
+        stmt.setDate(4, new java.sql.Date(date.getTime))
+        stmt.executeUpdate()
+      }
+
+    if (historyUpdated == 0) {
+      // there should only be one thread trying to due this at a time since this is only called in the SecondaryWatcher
+      try {
+        using(conn.prepareStatement(
+          """INSERT INTO secondary_metrics_history(store_id, dataset_system_id, date, total_size)
+                  VALUES (?, ?, ?, ?)
+        """)) { stmt =>
+          stmt.setString(1, storeId)
+          stmt.setDatasetId(2, datasetId)
+          stmt.setDate(3, new java.sql.Date(date.getTime))
+          stmt.setLong(4, metric.totalSizeBytes)
+          stmt.execute()
+        }
+      } catch {
+        case uv@PostgresUniqueViolation(_*) =>
+            // there should only be one thread trying to due this at a time since this is only called in the SecondaryWatcher
+            log.error("Inserting into secondary_metrics_history failed after updating no rows", uv)
+      }
+    }
+
+// TODO: replace this with using Postgres UPSERT functionality introduced in 9.5
```

Comment:
do you want to keep this whole commented-out section?

Reply:
Yes, I already did this for pg 9.6 before I found out we have 9.4 :(